WalletWasabi/Blockchain/Keys/HdPubKeyCache.cs

```diff
@@ -11,14 +11,16 @@ public class HdPubKeyCache : IEnumerable<HdPubKey>
 {
 	private Dictionary<Script, HdPubKey> HdPubKeysByScript { get; } = new();
 	private HashSet<HdPubKey> HdPubKeys { get; } = new();
-	private Dictionary<KeyPath, byte[]> ScriptBytesByKeyPath { get; } = new();
+	private Dictionary<HdPubKey, byte[]> ScriptBytesByHdPubKey { get; } = new();
```

Comment:
This doesn't look good. It is always a bad idea to use really big and complex mutable objects as keys in a dictionary but in this case the problem here is that the `HdPubKey` is exposed only to have access to the `LatestSpendingHeight` property. So, it would be safer to revert this change and replace the Value with an accessor to the property:

```suggestion
	private Dictionary<KeyPath, (byte[], HdPubKey)> ScriptBytesByHdPubKey { get; } = new();
```

or
```suggestion
	private Dictionary<KeyPath, (byte[], Func<Height>)> ScriptBytesByHdPubKey { get; } = new();
```